python/kfserving/kfserving/api/kf_serving_client.py

```diff
@@ -325,3 +326,55 @@ def delete(self, name, namespace=None):
             raise RuntimeError(
                 "Exception when calling CustomObjectsApi->delete_namespaced_custom_object:\
                  %s\n" % e)
+
+
+    def is_isvc_ready(self, name, namespace=None): #pylint:disable=inconsistent-return-statements
+        """
+        Check if the inference service is ready.
+        :param name: inference service name
+        :param namespace: defaults to current or default namespace
+        :return:
+        """
+        kfsvc_status = self.get(name, namespace=namespace)
+        status = 'Unknown'
+        for condition in kfsvc_status['status'].get('conditions', {}):
+            if condition.get('type', '') == 'Ready':
+                status = condition.get('status', 'Unknown')
+                return status.lower() == "true"
+        return False
+
+
+    def wait_for_ready(self, name, namespace=None, #pylint:disable=too-many-arguments
+                       timeout_seconds=600,
+                       polling_interval=10,
+                       debug=False):
+        """
+        Wait for inference service ready, print out the controller logs if timeout and debug=True.
+        :param name: inference service name
+        :param namespace: defaults to current or default namespace
+        :param timeout_seconds: timeout seconds for wait, default to 600s
+        :param polling_interval: The time interval to poll status
+        :param debug: If debug is True, print out the kfserving controller logs if timeout.
+        :return:
+        """
+        for _ in range(round(timeout_seconds/polling_interval)):
+            time.sleep(polling_interval)
+            if self.is_isvc_ready(name, namespace=namespace):
+                return
+
+        if debug:
```

Comment:
if this is user facing API not sure if we should print out the controller logs, I suggest printing all the inference service conditions.

Reply:
@yuzisun OK, I think we can print the inference service that includes status/conditions and other infomations.

Reply:
Can we remove the debug flag and print as Exception message ? If the debug flag is not on then the `RuntimeError` still says `See above log` not there is no log.

Reply:
@yuzisun Updated, thanks!